language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/LetVariableDeclarationNodeContext.java

```diff
@@ -58,38 +58,37 @@ public LetVariableDeclarationNodeContext() {
     public List<LSCompletionItem> getCompletions(BallerinaCompletionContext context, LetVariableDeclarationNode node) {
         List<LSCompletionItem> completionItems = new ArrayList<>();
         int cursor = context.getCursorPositionInTree();
-        if (node.typedBindingPattern().typeDescriptor().textRange().endOffset() >= cursor) {
-            /*
-            Covers the following context
-            eg: let va<cursor>
-                let int a = b, f<cursor>
-             */
-            completionItems.addAll(this.getTypeDescContextItems(context));
-            completionItems.add(new SnippetCompletionItem(context, Snippet.KW_VAR.get()));
-            for (LSCompletionItem lsCItem : completionItems) {
-                CompletionItem completionItem = lsCItem.getCompletionItem();
-                completionItem.setSortText(SortingUtil.genSortTextForTypeDescContext(context, lsCItem));
+        TypeDescriptorNode typeDescriptor = node.typedBindingPattern().typeDescriptor();
+        if (typeDescriptor.textRange().endOffset() >= cursor) {
+            if (typeDescriptor.kind() == SyntaxKind.QUALIFIED_NAME_REFERENCE) {
+                /*
+                Covers the following context
+                eg: let var x = <cursor>
+                    let var x = h<cursor>
+                    let var x = mod1:<cursor>
+                */
+                QualifiedNameReferenceNode qNameRef = (QualifiedNameReferenceNode) typeDescriptor;
+                List<Symbol> exprEntries = QNameRefCompletionUtil.getExpressionContextEntries(context, qNameRef);
+
+                completionItems.addAll(this.getCompletionItemList(exprEntries, context));
+                this.sort(context, node, completionItems);
```

Comment:
Shall we add an early return here and get rid of the else block?

Reply:
Done